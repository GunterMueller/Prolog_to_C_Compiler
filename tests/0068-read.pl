main :-
	see('tests/0060-crypta.pl'),
	repeat, read(X), write(X), nl, X == end_of_file.



