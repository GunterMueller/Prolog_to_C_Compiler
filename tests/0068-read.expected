solve(_1607,_1693,_2237) :- in_form(_1607,_2237,_2070,_2101),in_form(_1693,_2237,_2138,_2169),reverse(_2237,_1874),add_parallel(_2070,_2138,0,_1874,[1,2,3,4,5,6,7,8,9,0]),out_form(_2070,_2206,_2101),out_form(_2138,_2222,_2169),report(_2206,_2222,_2237)
add_parallel([_3313],[_3329],_3344,[_3359],_3388) :- add(_3313,_3329,_3344,_3359,0,_3388,_3403)
add_parallel([_4824|_4964],[_4840|_4980],_4855,[_4870|_5010],_4900) :- add(_4824,_4840,_4855,_4870,_4995,_4900,_5025),add_parallel(_4964,_4980,_4995,_5010,_5025)
add(_6820,_7073,_7172,_7373,_7459,_6836,_7419) :- digit(_6820,_6836,_6904),digit(_7073,_6904,_7404),_7475 is _6820+_7073+_7172,_7389 is _7475 rem 10,check(_7373,_7389,_7404,_7419),calc_carry(_7459,_7475)
check(_8497,_8312,_8342,_8342) :- integer(_8497),_8312 is _8497
check(_9260,_9396,_9308,_9323) :- var(_9260),select(_9396,_9308,_9323),_9260 is _9396
calc_carry(1,_9908) :- _9908 >= 10
calc_carry(0,_10446) :- _10446 < 10
digit(_11062,_10978,_10978) :- integer(_11062)
digit(_11709,_11725,_11740) :- var(_11709),select(_11709,_11725,_11740)
in_form(_12494,_12510,_12596,_12540) :- pad(_12494,_12510,_12580,_12540),reverse(_12580,_12596)
pad(_13871,_13649,_13887,_13902) :- length(_13871,_13816),length(_13649,_13665),_13902 is _13665-_13816,extend(_13871,_13887,_13902,0)
extend(_14448,_14448,0,_14477) :- !
extend(_15455,_15694,_15324,_15724) :- append([_15724],_15455,_15678),_15709 is _15324-1,extend(_15678,_15694,_15709,_15724)
out_form(_16633,_16784,_16799) :- reverse(_16633,_16699),extend(_16768,_16699,_16799,0),extend(_16768,_16784,_16799, )
report(_17722,_17754,_17890) :- printlst(_17722),printlst(_17754),length(_17890,_17856),printchr(-,_17856),printlst(_17890),nl
printlst(_18637) :- nl,tab(10),putlt(_18637)
printchr(_19257,_19273) :- nl,tab(10),putch(_19257,_19273)
putlt([])
putlt([_19949|_19981]) :- display(_19949),putlt(_19981)
putch(_20334,0) :- !
putch(_21228,_20944) :- display(_21228),_21244 is _20944-1,putch(_21228,_21244)
main :- main1,main2,main3
main1 :- solve([_22617,_22871,_22723,_22768,_22768],[_22871,_22723,_22811,_22795,_22768],[_22795,_22811,_22826,_22841,_22856,_22871])
main2 :- solve([_23622,_23816,_23801,_23668],[_23770,_23786,_23728,_23816],[_23770,_23786,_23801,_23816,_23831])
main3 :- solve([_24948,_24991,_24798,_24918,_24933,_24948],[_24872,_25021,_25036,_24918,_24933,_24948],[_25036,_24991,_25006,_25021,_25036,_25051])
end_of_file
